### PR TITLE
[WIP] shipit-workflow: Initial Ship-it API

### PR DESCRIPTION
**Pull request by rail**
This is the initial API that I would like to get some feedback on.

### CODE REVIEW

File: src/shipit_workflow/shipit_workflow/api.yml [hunk 1]
```diff
@@ -1,19 +1,125 @@
...
+    get:
+      summary: List releases
+      operationId: shipit_workflow.api.list_releases
+      parameters:
+      - in: query
+        name: full
+        type: boolean
+      responses:
+        200:
+          description: all good
```
Comment (garbas) [inline src/shipit_workflow/shipit_workflow/api.yml]: schema of response is not defined

File: src/shipit_workflow/shipit_workflow/api.yml [hunk 2]
```diff
@@ -1,19 +1,125 @@
...
+      parameters:
+      - name: name
+        in: path
+        required: true
+        type: string
+        description: release name
       responses:
         200:
-          description: Nothing to say...
+          description: all good
```
Comment (garbas) [inline src/shipit_workflow/shipit_workflow/api.yml]: each response should have schema defined.

File: src/shipit_workflow/shipit_workflow/api.yml [hunk 3]
```diff
@@ -1,19 +1,125 @@
...
+        required: true
+        type: string
+        description: release name
+      - name: phase
+        in: path
+        required: true
+        type: string
+      responses:
+        200:
+          description: all good
```
Comment (garbas) [inline src/shipit_workflow/shipit_workflow/api.yml]: each response should have schema defined.

File: src/shipit_workflow/shipit_workflow/api.yml [hunk 4]
```diff
@@ -1,19 +1,125 @@
...
+        type: string
+        description: release name
+      - name: phase
+        in: path
+        required: true
+        type: string
+        required: true
+      responses:
+        409:
+          description: In case it was submitted already
```
Comment (garbas) [inline src/shipit_workflow/shipit_workflow/api.yml]: each response should have schema defined.

Comment (rail) [reply src/shipit_workflow/shipit_workflow/api.yml]: I'll update the swagger definition.

**Pull request merged**